test/media/drm_engine_unit.js

```diff
@@ -610,6 +621,92 @@ describe('DrmEngine', () => {
             })],
             distinctiveIdentifier: 'required',
             persistentState: 'required',
+            initDataTypes: ['cenc'],
+          })]);
+    });
+
+    it('should silently try PlayReady recommendation keySytem', async () => {
+      shaka.util.Platform.isEdge = () => true;
+
+      // Leave only one drmInfo
+      manifest = shaka.test.ManifestGenerator.generate((manifest) => {
+        manifest.addVariant(0, (variant) => {
+          variant.addVideo(1, (stream) => {
+            stream.encrypted = true;
+            stream.addDrmInfo('com.microsoft.playready');
+          });
+          variant.addAudio(2, (stream) => {
+            stream.encrypted = true;
+            stream.addDrmInfo('com.microsoft.playready');
+          });
+        });
+      });
+
+      setRequestMediaKeySystemAccessSpy([
+        'com.microsoft.playready',
+      ]);
+      config.servers = {
+        'com.microsoft.playready': 'https://com.microsoft.playready/license',
+      };
+      config.advanced['com.microsoft.playready'] = {
+        audioRobustness: 'good',
+        videoRobustness: 'really_really_ridiculously_good',
+        serverCertificate: null,
+        sessionType: '',
+        individualizationServer: '',
+        distinctiveIdentifierRequired: false,
+        persistentStateRequired: true,
+      };
+
+      drmEngine.configure(config);
+
+      const variants = manifest.variants;
+      await drmEngine.initForPlayback(variants, manifest.offlineSessionIds);
+
+      const expectedConfig = jasmine.objectContaining({
+        audioCapabilities: [jasmine.objectContaining({
+          robustness: 'good',
+        })],
+        videoCapabilities: [jasmine.objectContaining({
+          robustness: 'really_really_ridiculously_good',
+        })],
+        persistentState: 'required',
+        initDataTypes: ['cenc'],
+      });
```

Comment:
Are the `config.advanced` and `expectedConfig` fields needed?  Can we just use the defaults?  This should reduce the amount of things tested to just verifying the two calls to `requestMediaKeySystemAccessSpy`.  You can use `jasmine.any(Array)` as the second argument to `toHaveBeenCalledWith` to ignore it.